django/db/backends/base/schema.py

```diff
@@ -118,6 +118,17 @@ def __exit__(self, exc_type, exc_value, traceback):
 
     # Core utility functions
 
+    def prepare_param(self, node):
+        compiler = self.connection.ops.compiler("SQLCompiler")(query=None, connection=self.connection, using=None)
+        vendor_impl = getattr(node, 'as_' + self.connection.vendor, None)
+        if vendor_impl:
+            sql, params = vendor_impl(compiler, self.connection)
+        elif hasattr(node, 'as_sql'):
+            sql, params = node.as_sql(compiler, self.connection)
+        else:
+            sql, params = None, (node,)
+        return sql, tuple(params)
```

Comment:
We expect compileable to always have an `as_sql` method. e.g. see `effective_default`'s implementation.

```suggestion
        if not hasattr(node, 'as_sql'):
            return None, (node,)
        compiler = self.connection.ops.compiler('SQLCompiler')(
            query=None, connection=self.connection, using=None
        )
        return compiler.compile(node)
```